### PR TITLE
Follow first-level symlinks.

### PR DESCRIPTION
**Pull request by maltoe**
Hello thoughtbot team,

in a project using json_matcher we wanted to be able to symlink a directory into the `JsonMatcher.schema_root`, unfortunately the `Matcher.build_and_populate_document_store` method uses a glob pattern that doesn't follow symlinks. Might be a bit of an edge case, but allowing to traverse first-level symlinks is relatively trivial.

However, testing is not, or at least the factory-based test setup makes it a bit harder, in my eyes. So if you say this change is worth it, I would come up with a test for it, maybe with a little hint on how to put a symlinked path into the factories. Or maybe you say this doesn't need a test? :)

cheers,
malte

### CODE REVIEW

File: lib/json_matchers/matcher.rb [hunk 1]
```diff
@@ -31,7 +31,9 @@ def validator
     def build_and_populate_document_store
       document_store = JsonSchema::DocumentStore.new
 
-      Dir.glob("#{JsonMatchers.schema_root}/**/*.json").
+      # follow one symlink and direct children
+      # http://stackoverflow.com/questions/357754/can-i-traverse-symlinked-directories-in-ruby-with-a-glob
+      Dir.glob("#{JsonMatchers.schema_root}/**{,/*/**}/*.json").
```
Comment (seanpdoyle) [inline lib/json_matchers/matcher.rb]: Thanks for this contribution!

In my experience, I've found that code comments with URLs like this can be really helpful to keep close to the code that might strike the next developer as an oddity.

With that in mind, how do you feel about extracting out a private method named in a way that communicates the intent behind this decision?

One benefit is that a well-named private method is more likely to survived future refactorings. Personally, I feel that it also serves as an anchor for code comments like this.

**Comment by seanpdoyle** [comment]
In addition to code-level feedback, could you elaborate on the details of this PR in the commit message?

I think what you've mentioned in the PR description would serve us well enough.

Thanks for the contribution!

**Comment by maltoe** [comment]
Hey @seanpdoyle,

I extended the commit message and extracted the `Glob.()` part into a private method as you suggested. Not sure if the message name communicates the "intent" though, but rather the mechanic :) The intent makes it very difficult to be put in a method name, in my opinion.

Thank you for your review!

File: lib/json_matchers/matcher.rb [hunk 2]
```diff
@@ -31,13 +31,19 @@ def validator
...
+      traverse_schema_root_with_first_level_symlinks.
         map { |path| Pathname.new(path) }.
         map { |schema_path| Parser.new(schema_path).parse }.
         map { |schema| document_store.add_schema(schema) }.
         each { |schema| schema.expand_references!(store: document_store) }
 
       document_store
     end
+
+    def traverse_schema_root_with_first_level_symlinks
```
Comment (seanpdoyle) [inline lib/json_matchers/matcher.rb]: This is great! Thank you

**Comment by seanpdoyle** [comment]
Could you also update this PR to mention the change in the [./NEWS.md](https://github.com/thoughtbot/json_matchers/blob/master/NEWS.md) file (and link to this PR)?

**Comment by ammancilla** [comment]
Hi guys,

Any plans on getting this merged and released at some point?

Cheers and thanks for this!
